ddraw/IDirect3DDeviceX.cpp

```diff
@@ -2453,11 +2542,98 @@ HRESULT m_IDirect3DDeviceX::DrawIndexedPrimitive(D3DPRIMITIVETYPE dptPrimitiveTy
 		}
 		else
 		{
+			const UINT stride = GetVertexStride(dwVertexTypeDesc);
+
+			// Handle PositionT
+			if((dwVertexTypeDesc & D3DFVF_XYZRHW) != 0 && Config.DdrawConvertHomogeneousW)
+			{
+				if(!Config.DdrawConvertHomogeneousToWorld)
+				{
+					UINT8 *vertex = (UINT8*)lpVertices;
+
+					for (UINT x = 0; x < dwVertexCount; x++)
+					{
+						float *pos = (float*) vertex;
+
+						pos[3] = 1.0f;
+
```

Comment:
This changes the contents of the source geometry. Generally it's not a good idea to do this, since that memory might reside in read-only memory, might be hashed, or used for other purposes. Such use cases will break. Better use an intermediate copy, just like the else case, except it can be copied over fully before modifications the single component

Reply:
This has been commented out.